### PR TITLE
feat: open AWS SSO login in default browser

### PR DESCRIPTION
**Pull request by hoegertn**
**Changelog**

Open the AWS SSO login in the default browser to support password managers and existing sessions

### CODE REVIEW

**Comment by andreacavagna01** [comment]
There are still some problems with this feature:
- the app saves an Invalid `EXPIRATION_TIME` and saves `Invalid Date`.
- By doing this, the login will be simplified but, it prevents us to make more feature, like the multiple AWS SSO, (because the current window opening in Electron grants us Context Isolation).
What are the benefits of your Feature?

**Comment by hoegertn** [comment]
I am not sure how this change influences this as it only defers the token resolution but I did not change the flow itself.

The benefit is that the auth is delegated to your normal browser that knows who you are, has additional factors, trusts your AD etc. So you can login easier.

Currently, I need to enter username and password all the time and then use my 2FA because the browser is not recognized. For enterprise setups this includes redirects to AzureAD, login there with different factors etc.

The default browser already knows all this stuff.

File: src/app/integrations/providers/aws-sso.service.ts [hunk 1]
```diff
@@ -144,6 +110,12 @@ export class AwsSsoService extends NativeService {
 
     return fromPromise(this.ssooidc.createToken(createTokenRequest).promise()).pipe(
       catchError((err) => {
+        if (err.code === 'AuthorizationPendingException') {
+          // Retry after 5 seconds if token is not yet ready
+          return timer(5000).pipe(
```
Comment (andreacavagna01) [inline src/app/integrations/providers/aws-sso.service.ts:115]: Is polling the best integration we can do with the external opening window, we are some kind of losing control of what happens in the flow.

Should be a good idea to add a maximum number of retry to manage this integration I think.

Comment (hoegertn) [reply src/app/integrations/providers/aws-sso.service.ts:115]: As I understand it the request times out on AWS side and the error will change.

File: src/app/integrations/providers/aws-sso.service.ts [hunk 2]
```diff
@@ -144,6 +110,12 @@ export class AwsSsoService extends NativeService {
...
+            switchMap(() => this.generateSSOToken(authorizeIntegrationResponse))
+          );
+        }
         return throwError(`AWS SSO token creation error: ${err.toString()}`);
       }),
       switchMap((createTokenResponse: any) => {
```
[the remark this answers is not in the document]
Comment (hoegertn) [reply src/app/integrations/providers/aws-sso.service.ts:121]: Ok, any hints? I might not know enough about RX for this.

File: src/app/integrations/providers/aws-sso.service.ts (hunk 1 above)
Comment (andreacavagna01) [reply src/app/integrations/providers/aws-sso.service.ts:115]: You are right, this use-case should be already managed

**Comment by hoegertn** [comment]
@andreacavagna01 Hi, can you elaborate on the roadmap? Is this going to be reintroduced soon?

**Comment by lightbody** [comment]
Hi there -- I'm a bit confused. I'm running 0.5.3 but I don't see any option to force Leapp to run the authentication process for AWS SSO through to my normal browser. Am I missing something or is this feature not released yet?

**Comment by ericvilla** [comment]
Hi @lightbody, as you said this feature is not available in 0.5.3

We released this feature in [v0.4.7](https://github.com/Noovolari/leapp/releases/tag/v0.4.7) release but, for stability reasons, we decided to remove it since [v0.5.1](https://github.com/Noovolari/leapp/releases/tag/v0.5.1)

So far, we need to understand how to introduce it in a robust way, enhancing the current in-app experience.

**Comment by andreacavagna01** [comment]
Hi, the previous merged version has needed some improvement before being released. 
We are working hard to make the project follow the precise specifications. 

This specification will be released in the next weeks and the project will be refactored following those rules.

By doing so, we can add Unit Test on the core logic of Leapp, and then all the other features previously implemented will be released.

For this specific feature, this will be introduced as an option so both users can still use Leapp, the one that needs authentication inApp and the one that wants authentication in Browser.

All this will be released soon, please stay tuned and join our [community](https://join.slack.com/t/noovolari/shared_invite/zt-opn8q98k-HDZfpJ2_2U3RdTnN~u_B~Q) if you want to stay updated with the last news on the app.

I'm here for any further question

**Comment by lightbody** [comment]
Thank you for the clarification, @ericvilla @andreacavagna01

**Pull request merged**